geoh5py/ui_json/validation.py

```diff
@@ -51,17 +46,54 @@ class InputValidation:
 
     def __init__(
         self,
-        validations: dict[str, Any] | None,
+        validators: dict[str, BaseValidator] = None,
+        validations: dict[str, Any] | None = None,
         ignore_list: tuple = (),
         ignore_requirements: bool = False,
         workspace: Workspace = None,
+        ui_json: dict[str, Any] = None,
     ):
-        self._validators: dict[str, BaseValidator] = {}
+        self.validations: dict[str, Any] | None = validations
+        self.ui_json: dict[str, Any] = ui_json
+        self.validators: dict[str, BaseValidator] = validators
         self.workspace: Workspace | None = workspace
-        self.validations = validations
         self.ignore_list: tuple = ignore_list
         self.ignore_requirements: bool = ignore_requirements
 
+    @property
+    def ui_json(self):
+        return self._ui_json
+
+    @ui_json.setter
+    def ui_json(self, val):
+        if val is None:
+            self._inferred_validations = {}
+        else:
+            self._inferred_validations = self.infer_validations(val)
+        self._ui_json = val
+
+    @property
+    def validations(self):
+        return dict(self._inferred_validations, **self._validations)
+
+    @validations.setter
+    def validations(self, val: dict[str, Any]):
+        self._validations = val
+
+    @property
+    def validators(self):
+        return self._validators
+
+    @validators.setter
+    def validators(self, val):
+        v = {} if val is None else val
```

Comment:
We should probably check for `isinstance(val, "BaseValidator")`, otherwise the merge with required_validators won't make much sense.

Reply:
val isn't expected to be a BaseValidator here, but a dictionary of validators.  How about typing this with `val: dict[str, BaseValidator]`

Reply:
Oh yep, sure. That makes more sense.

Reply:
Done